hbase-server/src/test/java/org/apache/hadoop/hbase/regionserver/TestSyncTimeRangeTracker.java

```diff
@@ -84,23 +86,23 @@ public void run() {
     assertTrue(trr.getMin() == 0);
   }
 
-  static class RandomTestData {
-    private long[] keys = new long[NUM_KEYS];
-    private long min = Long.MAX_VALUE;
-    private long max = 0;
+ static class RandomTestData {
+    private final AtomicLongArray keys = new AtomicLongArray(NUM_KEYS);
```

Comment:
This is the 'fix'? Using atomic array. Sounds good.

Reply:
But this array will only be accessed by one thread? Why do we need to use AtomicLongArray?

Reply:
Basically I ran this test a lot before and after these changes to try and confirm a fix and this is what seemed to work.

Technically, even if a single thread accesses a data structure, if it is a different thread than the one that created it, you still need to publish objects correctly for multiple threads - concurrency is not required. So they need a mem barrier or to be final or to be effectively final or a 32-bit primitive(I think?).

Reply:
Yeah, so AtomicLongArray to ensure we see an up to date value.